### PR TITLE
BaSP-MR-8:Admin methods get update

### PR DESCRIPTION
**Pull request by Eugeniagassman**
In this code are the methods getAll, getById, update and the validation of update. 
In the GetById method first it verifies if a data with that id exists, if it does not exist it gives error, and if it exists it returns a message with the found id.
 In the method update first searches by id to edit, and then it is fixed which is the data that you want to modify, those data are validated and if they pass the validation the data are modified. 
 Also in the update method it is verified that if a new data is passed, if a new data is not passed it does not change the previous one.

### CODE REVIEW

File: src/controllers/admins.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      if (!result) {
+        return res.status(404).json({
+          message: `Admin with id: ${id} was not found`,
+          error: true,
+        });
+      }
+      return res.status(200).json(result);
+    })
+    .catch((error) => {
+      res.status(400).json(error);
```
Comment (fernandofernandezuy) [inline src/controllers/admins.js:73]: The responses must have the same structure in the code, here is the example they gave in the weekly task
 
    createdMember.save((error, client) => {
       if (error) {
           return res.status(400).json({
               message: error, // error.toString()
               data: undefined,
               error: true
           });
       }
       return res.status(201).json({
           message: 'Member created',
           data: client,
           error: false
       });
   });

File: src/controllers/admins.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      res.status(400).json({
+        message: 'An error ocurred',
+        error,
+      });
+    });
+};
+
+const getAdminById = (req, res) => {
+  const { id } = req.params;
+  Admin.findById(id, 'firstName')
```
Comment (Juanoauil) [inline src/controllers/admins.js:22]: Why do you return only first name?

File: src/controllers/admins.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    .catch((error) => {
+      res.status(400).json({
+        message: 'An error ocurred',
+        error,
+      });
+    });
+};
+
+const getAdminById = (req, res) => {
+  const { id } = req.params;
```
Comment (Juanoauil) [inline src/controllers/admins.js:21]: Please check if id exist before search in DB

File: src/controllers/admins.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    .catch((error) => {
+      res.status(400).json({
+        message: 'An error ocurred',
+        error,
+      });
+    });
+};
+
+const updateAdmin = (req, res) => {
+  const { id } = req.params;
```
Comment (Juanoauil) [inline src/controllers/admins.js:46]: Same here

Comment (Eugeniagassman) [reply src/controllers/admins.js:59]: resolved

**Review by fernandofernandezuy — APPROVED** [review]
Looks good to me!

**Review by GaloDurante — APPROVED** [review]

File: src/controllers/admins.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    res.status(400).json({
+      message: `Id: ${id} invalid, try a valid id`,
+      data: undefined,
+      error: true,
+    });
+  } else {
+    Admin.findById(id, 'firstName lastName dni phone email city password')
+      .then((admins) => {
+        if (!admins) {
+          res.status(404).json({
```
Comment (Juanoauil) [inline src/controllers/admins.js:35]: add return in fuction

File: src/models/Admin.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  firstName: {
+    type: String,
+    require: true,
+  },
+  lastName: {
+    type: String,
+    require: true,
+  },
+  dni: {
+    type: Number,
```
Comment (Juanoauil) [inline src/models/Admin.js:14]: Dni must be string

File: src/models/Admin.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    require: true,
+  },
+  phone: {
+    type: Number,
```
Comment (Juanoauil) [inline src/models/Admin.js:18]: Same that dni

Comment (Juanoauil) [reply src/models/Admin.js:18]: Fix joi validations too

File: src/validations/admins.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  if (number.toString().length === 10) {
+    return number;
+  } return helpers.message('Phone must be 10 numbers');
+};
+
+const validateAdminUpdate = (req, res, next) => {
+  const adminValidation = Joi.object({
+    firstName: Joi.string().min(3).max(15),
+    lastName: Joi.string().min(3).max(15),
+    dni: Joi.number().custom((number, helpers) => (numberDNILength(number, helpers))),
```
Comment (Juanoauil) [inline src/validations/admins.js:19]: If we set a string data, we do not need to set a custom validation because that make with string length

**Review by Juanoauil — APPROVED** [review]

**Pull request merged**